pyccl/nl_pt/ept.py

```diff
@@ -0,0 +1,700 @@
+            extrap_order_hik (int): extrapolation order to be used on
+                k-values above the maximum of the splines. See
+                :class:`~pyccl.pk2d.Pk2D`.
+
+        Returns:
+            :class:`~pyccl.pk2d.Pk2D`: PT power spectrum.
+        """
+        if not (kind in self._pk_valid):
+            # Reverse order and check again
+            kind_reverse = ':'.join(kind.split(':')[::-1])
+            if not (kind_reverse in self._pk_valid):
+                raise ValueError(f"Pk template {kind} not valid")
+            kind = kind_reverse
+        pk_name = self._pk_alias[kind]
+
+        if return_ia_bb and (pk_name in ['c2:c2', 'c2:cdelta',
+                                         'cdelta:cdelta']):
+            pk_name += '_bb'
+
+        # If already built, return
+        if pk_name in self._pk2d_temp:
+            return self._pk2d_temp[pk_name]
+
+        # Construct power spectrum array
+        s4 = 0.
+        if pk_name == 'm:m':
+            pk = self.pk_b1
+        elif pk_name == 'm:b2':
+            pk = 0.5*self._g4[:, None]*self.dd_bias[2][None, :]
+        elif pk_name == 'm:b3nl':
+            pk = 0.5*self._g4[:, None]*self.dd_bias[8][None, :]
+        elif pk_name == 'm:bs':
+            pk = 0.5*self._g4[:, None]*self.dd_bias[4][None, :]
+        elif pk_name == 'm:bk2':
+            pk = 0.5*self.pk_bk*(self.k_s**2)[None, :]
+        elif pk_name == 'm:c2':
+            pk = self._g4[:, None] * (self.ia_mix[0]+self.ia_mix[1])[None, :]
+        elif pk_name == 'm:cdelta':
+            pk = self._g4[:, None] * (self.ia_ta[0]+self.ia_ta[1])[None, :]
+        elif pk_name == 'b2:b2':
+            if self.fastpt_par['sub_lowk']:
+                s4 = self.dd_bias[7][:, None]
+            pk = 0.25*self._g4[:, None]*(self.dd_bias[3][None, :] - 2*s4)
+        elif pk_name == 'b2:bs':
+            if self.fastpt_par['sub_lowk']:
+                s4 = self.dd_bias[7][:, None]
+            pk = 0.25*self._g4[:, None]*(self.dd_bias[5][None, :] - 4*s4/3)
+        elif pk_name == 'bs:bs':
+            if self.fastpt_par['sub_lowk']:
+                s4 = self.dd_bias[7][:, None]
+            pk = 0.25*self._g4[:, None]*(self.dd_bias[6][None, :] - 8*s4/9)
+        elif pk_name == 'c2:c2':
+            pk = self._g4[:, None] * self.ia_tt[0][None, :]
+        elif pk_name == 'c2:c2_bb':
+            pk = self._g4[:, None] * self.ia_tt[1][None, :]
+        elif pk_name == 'c2:cdelta':
+            pk = self._g4[:, None] * self.ia_mix[2][None, :]
+        elif pk_name == 'c2:cdelta_bb':
+            pk = self._g4[:, None] * self.ia_mix[3][None, :]
+        elif pk_name == 'cdelta:cdelta':
+            pk = self._g4[:, None] * self.ia_ta[2][None, :]
+        elif pk_name == 'cdelta:cdelta_bb':
+            pk = self._g4[:, None] * self.ia_ta[3][None, :]
+        elif pk_name == 'zero':
+            # If zero, store None and return
+            self._pk2d_temp[pk_name] = None
+            return None
```

Comment:
Just `return`, no `None`.
If you want to be extra-spicy, since `setattr` returns None, you can combine both lines into 1:
`return self._pk2d_temp[pk_name] = None`. Kind of like Julia.

Reply:
This doesn't work (syntax error). Leaving as is to be less cryptic

Reply:
I prefer to be explicit.